README.md
# ![LightLogo](./docs/static/header_logo.svg)

![build](https://github.com/skilkis/tudelft-light/workflows/build/badge.svg)

A lightweight XeLaTeX Template for TU-Delft assignments and reports.

## Installation

This template can be used right away by either downloading the released files
or by embedded the template itself as a Git submodule. For the latter, a
`.gitmodules` file can be defined as explained by the instructions in the
[submodule](https://github.com/skilkis/tudelft-light-template).

## Highlights

### TikZ Cover Page

| Dark Theme | Light Theme |
| --- | --- |
| <img src="./docs/static/cover_dark.png" width="450px"> | <img src="./docs/static/cover_light.png" width="450px">

The cover pages above can be created quickly with only a single command!

```latex
\makecover[theme=light, fill opacity=0.1]
```

The `\makecover` macro is also highly customizable:

```latex
\makecover[
    theme=light,
    fill opacity=0.1,
    image scale=1,
    image angle=0,
    image xshift=0,
    image yshift=0,
    image file=tudelft-light/images/background_light.pdf
    title fontsize=72pt,
]
```

### Inner Title Page

An inner title page contains further metadata about the report/project such as
the course code, author(s), supervisor(s), and an optional short abstract.
This metadata is also automatically added to the metadata of the compiled PDF.

<img src="./docs/static/cover_inner.png" width="100%">

An included macro makes it easy to format multiple authors, student numbers,
and supervisors:

```latex
\author{John Doe | 0000001, Jane Doe | 0000002}
\supervisor{Dr. Jan Jansen}
```

### Code Listings using `minted`

To use this functionality the `Pygments` package has to be available on
the system `PATH`. Once this requirement is satisfied, adding code listings
for virtually any language is a piece of cake.

<img src="./docs/static/listing.png" width="100%">


CHANGELOG.md
# Changelog

All notable changes to this project are documented in this file.

The format is based on [Keep a Changelog], and this project adheres to
[Semantic Versioning].

## [2.4.0] - 2021-03-21

Big release, bringing quality of life improvements to the template as well as
important fixes! One of the most notable is the incorporation of `book.cls`
style `\frontmatter` and `\backmatter` macros that reduces the `main.tex` file
down to only a few lines!

```latex
\documentclass[11pt,a4paper,reqno,fancyheader]{tudelft-light/tud-report}

\title[TU Delft Light]{TU Delft Light}
\subtitle{An Easy to Use \LaTeX\ Template}
\author{John Doe | 0000001, Jane Doe | 0000002}  % {Name | Student Number}
\course{[TU0000] \LaTeX\ 101}
\supervisor{Dr. Jan Jansen}
\abstract{\input{examples/abstract.tex}}

\begin{document}
    \makecover[theme=light, fill opacity=0.1]
    \maketitle
    \frontmatter
        \tableofcontents
        \printnomenclature[50pt]
        \listoffigures
        \listoftables
    \mainmatter
        \include{examples/latex_elements}
        \printbibliography[title=References,heading=bibintoc]
        \begin{appendices}
            \include{examples/listing}
        \end{appendices}
\end{document}
```

### Added

- Ability to generate a formatted bibliography that appears directly
  in the Table of Contents with the `\printbibliography` command (Issue #13)
- Asymmetric margins when using the `twoside` option (Issue #25)
- `questions` environment for declaring research questions (Issue #27).
  Special thanks to @kswannet for providing the suggestion!
- Automated inclusion of list of figures and list of tables in
  table of contents using the `tocbibind` package. Refer to `main.tex` to
  see how to prevent duplicates in ToC due to this addition.
- Automated filtering of citations in captions using the `etoolbox` package
- `\frontmatter` and `\backmatter` macros to easily change between a roman
  numbered chapters such as the preface/summary to main content chapters
  numbered with arabic numerals. This greatly reduces the verbosity of the
  `main.tex` file.
- Support for ensuring that chapters start on odd-pages in when using the
  `twoside` mode (Issue #48). Blank pages that follow the current
  formatting are added with the text "This page intentionally left blank."
- Support for typesetting quotes with the `quoting` package

### Changed

- Hard-coded `\addbibresource` filename to use a glob pattern (Issue #38)
- Refactored imports to `tud-report.cls` file to make dependencies more
  explicit
- Hard-coded `\geometry` declaration for `twoside` option in `tud-report.cls`
  into conditional options that are passed to the `geometry` package (Issue
  #43)
- Cover and title page formatting to produce blank pages if `twoside` option
  is declared
- Listings to be numbered using the chapter number

### Removed

- `tud-packages.sty` package
- Unused `datatool`, `rotating`, `pdflscape`, `color`, `pdfpages` packages

### Fixed

- References appearing twice in fancy header (Issue #47)
- Nomenclature not compiling when using `latexmk` (Issue #50)
- Incorrect page numbers in ToC, caused by an incompatibility with the
  `notocite` package (Issue #37)

## [2.3.0] - 2021-02-18

### Added

- Ability to pass optional value argument to `\nomunit` (Issue #31)

### Fixed

- Failing compilation on TeXLive 2020 (Issue #34)

### Changed

- `\documentclass` to use `tudelft-light/tud-report` instead of
  `tudelft-light/report` (Issue #34)
- Use of `\import` to `\include` for adding report structure support in VS Code
  LaTeX Workshop and support the `\includeonly` macro (Issue #29)
- Bumped version of `tudelft-light-template` to
  [87c1708](https://github.com/skilkis/tudelft-light-template/commit/87c1708)

## [2.2.0] - 2020-12-01

### Added

- `title fontsize` option to the `\makecover` macro to allow users to
  format the cover page title font size (Issue #23)
- Remove hyphenation from cover page title text using the `hyphenat` package
- `title fontsize` argument to the [README] `\makecover` example

### Changed

- Bumped version of `tudelft-light-template` to [c2c6bcb]

### Fixed

- Grammar mistake in the Inner Title Page section of the [README]

## [2.1.0] - 2020-10-05

### Added

- SVG logo to the README as well as improved it with highlights and pictures
- Ability to automatically compile the template and create releases on tagged
  commits
- Automated plural suffix addition to the "Author" and "Supervisor" words
  in the inner title page

### Changed

- `tudelft-light` directory into a submodule to make it easier to embed
  a minimal template into other Git repositories

### Fixed

- Hardcoded supervisor name (Issue #14)

## [2.0.0] - 2020-02-20

An overhauled version that greatly increases usability. You can now generate
awesome cover pages and title pages easily!

### Added

- `\makecover` and `\maketitle` macros to generate the cover page and title
  page (Issue #1)
- Option to toggle display of a fancy header to the document class
- Use of the `import` package to fix build errors
- Units and Numbers section to the LaTeX example elements chapter
- Font files from [TU Delft Thesis Template]
- Light and dark vector cover images and logos
- Definition of TU Delft House Style fonts and colors
- Check to make sure that document is being compiled with [XeLaTeX]
- String labels for PDF page numbers (Issue #9)

### Changed

- Location of LaTeX example elements chapter and `abstract.tex` into
  `examples/` folder
- `usepackage` statements in the template to `RequirePackage` (Issue #4)
- Adopt correct usage of a document class file `.cls` instead of a package with
  multiple sub-packages to increase readability
- Usage of `listings` package in favor of `minted` (Issue #7)

### Removed

- `samplerefs.txt` and placed its contents into the [README]
- Unnecessary call to `\addcontentsline` macro by using the `intoc` option of
  the `nomencl` package
- Introduction chapter and old listing examples

examples/code/RunCase.m
% Copyright 2018 San Kilkis, Evert Bunschoten
% 
% Licensed under the Apache License, Version 2.0 (the "License");
% you may not use this file except in compliance with the License.
% You may obtain a copy of the License at
% 
%    http://www.apache.org/licenses/LICENSE-2.0
% 
% Unless required by applicable law or agreed to in writing, software
% distributed under the License is distributed on an "AS IS" BASIS,
% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
% See the License for the specific language governing permissions and
% limitations under the License.

classdef RunCase < handle
    %RUNCASE Summary of this class goes here
    %   Detailed explanation goes here
    
    properties
        aircraft;           % Aircraft instance with all parameters and vars
        x;                  % DesignVector object for fmincon & ease of use
        x_final;            % Optimized Design Vector
        converged = false;  % True if fmincon stopped w/o errors
        options;            % fmincon options struct
    end

    properties (SetAccess = private, GetAccess = public)
        cache = struct()    % Cache of Results & Constraints
        run_parallel        % Bool, True for machines with >= 4 cores
        iter_counter = 0    % Counts the number of function calls
        start_time          % datetime at the start of optimization
        end_time            % datetime at the end of optimization
        sim_time;           % Total Sim. Time at end of Optimization [s]
    end
    
    methods
        
        function obj = RunCase(aircraft_name, options)
            % Displaying welcome message
            type data\log\header.txt; fprintf('\n')
            
            % Constructing the specified aircraft
            obj.aircraft = aircraft.Aircraft(aircraft_name);
            obj.init_design_vector(); % Creating the design vector object

            % Augmenting options w/ OutputFnc
            options.OutputFcn = @obj.cache_optimValues;
            obj.options = options;
            obj.cache.results = []; % Results caching
            obj.cache.fmincon = []; % Solver caching
            obj.cache.const = [];   % Constraint caching
            obj.cache.time = [];    % Log of analysis time
        end

        function init_design_vector(obj)
            dv = @optimize.DesignVector;
            ac = obj.aircraft;
            obj.x = dv({'lambda_1', ac.lambda_1, 0, 1.25;...
                        'lambda_2', ac.lambda_2, 0.94, 1.25;...
                        'b', ac.b, 0.71, 1.06;...
                        'c_r', ac.c_r, 0.68, 1.15;...
                        'tau', ac.tau, 0.16, 2.5;...
                        'A_root', ac.A_root', 0.5, 1.2;...
                        'A_tip', ac.A_tip', 0.5, 1.2;...
                        'beta_root', ac.beta_root, 0, 1.7;...
                        'beta_kink', ac.beta_kink, -0.8, 3.2;...
                        'beta_tip', ac.beta_tip, -3.6, 3.6;...
                        % Get these values from first initial run
                        'A_L', ac.A_L, -1.5, 1.5;...
                        'A_M', ac.A_M, -1.5, 1.5;...
                        'W_w', ac.W_w, 0.6, 1.0;...
                        'W_f', ac.W_f, 0.6, 1.0;...
                        'C_d_w', ac.C_d_w, 0.8, 1.0});
        end

        function optimize(obj)
            obj.start_time = datetime(); tic;
            n_cores = feature('numcores');

            % Launching either in Parallel or Serial Execution
            try
                if n_cores >= 4
                    parpool(4)
                    obj.run_parallel = true;
                end
            catch
                obj.run_parallel = false;
                warning(['Parallel Processing Disabled ' ...
                         'or not Installed on Machine. Optimization '...
                         'will execute as a serial process!'])
            end

            [opt, ~] = fmincon(@obj.objective,...
                            obj.x.vector, [], [], [], [],...
                            obj.x.lb, obj.x.ub, @obj.constraints,...
                            obj.options);
                            
            obj.sim_time = toc;
            obj.x_final = opt;
            obj.end_time = datetime();
            obj.converged = true;
            obj.shutdown();
        end
        
        function [c, ceq] = constraints(obj, x)
            disp('Constraints')
            res = obj.fetch_results(x);
            Cons = optimize.Constraints(obj.aircraft, res, obj.x);
            c = Cons.C_ineq; ceq = Cons.C_eq;
            
            % Caching of constraints
            if isempty(obj.cache.const)
                obj.cache.const.c = c;
                obj.cache.const.ceq = ceq;
            else
                obj.cache.const.c(end+1, :) = c;
                obj.cache.const.ceq(end+1, :) = ceq;
            end
        end

        function fval = objective(obj, x)
            disp('Access from objective')
            res = obj.fetch_results(x);
            fval = res.W_f/obj.x.W_f_0;
        end
        
        function res = fetch_results(obj, x)
            if ~obj.x.isnew(x) && ~isempty(obj.cache.results)
                res = obj.cache.results(end);
            else
                disp('I asked for new runs')
                obj.x.vector = x; % Updates design vector w/ fmincon value
                obj.aircraft.modify(obj.x);
                obj.iter_counter = obj.iter_counter + 1;
                % Running Analysis Blocks
                if obj.run_parallel
                    tic;
                    spmd
                        if labindex == 1
                            temp = obj.run_aerodynamics();
                        elseif labindex == 2
                            temp = obj.run_structures();
                        elseif labindex == 3
                            temp = obj.run_loads();
                        elseif labindex == 4
                            temp = obj.run_performance();
                        end
                    end
                    t = toc;
                    fprintf('Parallel Process took: %.5f [s]\n', t)
                    res.C_dw = temp{1};
                    res.Struc = temp{2};
                    res.Loading = temp{3};
                    res.W_f = temp{4};
                else
                    tic;
                    res.C_dw = obj.run_aerodynamics();
                    res.Loading = obj.run_loads();
                    res.Struc = obj.run_structures();
                    res.W_f = obj.run_performance();
                    t = toc;
                end
                
                if isempty(obj.cache.results)
                    obj.cache.results = res;
                    obj.cache.time = t;
                else
                    obj.cache.results(end+1) = res;
                    obj.cache.time(end+1) = t;
                end
            end
        end
        
        function A = run_aerodynamics(obj)
            try
                Aero = aerodynamics.Aerodynamics(obj.aircraft);
                A = Aero.C_d_w;
            catch
                A.C_D_w = NaN;
            end
        end
        
        function L = run_loads(obj)
            try
                Loads = loads.Loads(obj.aircraft);
                L.M_distr = Loads.M_distr;
                L.L_distr = Loads.L_distr;
                L.Y_coord = Loads.Y_coord;
            catch
                L.M_distr = ones(length(obj.x.A_M),1) * NaN;
                L.L_distr = ones(length(obj.x.A_M),1) * NaN;
                L.Y_coord = NaN;
            end
        end
        
        function S = run_structures(obj)
            try
               Structures = structures.Structures(obj.aircraft);
               S.W_w = Structures.W_w;
               S.V_t = Structures.V_t;
            catch
                S.W_w = NaN;
                S.V_t = NaN;
            end
        end
        
        function P = run_performance(obj)
            try
                perf = performance.Performance(obj.aircraft);
                P = perf.W_fuel;
            catch
                P.W_fuel = NaN;
            end
        end

        function stop = cache_optimValues(obj, x, optimValues, state)
            stop = false;
            o = optimValues; 
            switch state
                case 'init'
                    % hold on
                    obj.cache.fmincon = o;
                    obj.cache.fmincon.x = x;
                case 'iter'
                    % Concatenate current point and objective function
                    % value with history. x must be a row vector
                    history = obj.cache.fmincon;
                    temp.fval = [history.fval, o.fval];
                    temp.x = [history.x, x];
                    
                    % Gradient caching of fmincon
                    temp.gradient = [history.gradient,...
                                     o.gradient];
                    
                    % Optimality caching of fmincon
                    temp.firstorderopt = [history.firstorderopt,...
                                          o.firstorderopt];
                                      
                    temp.iteration = [history.iteration, o.iteration];
                    temp.funccount = [history.funccount, o.funccount];
                    obj.cache.fmincon = temp;

                case 'done'
                    % hold off
                otherwise
            end
        end

        function shutdown(obj)
            if obj.run_parallel
                % Shutting Down Parallel Pool
                poolobj = gcp('nocreate');
                delete(poolobj);
            end
            obj.end_time = datetime();
            if isempty(obj.sim_time)
                obj.sim_time = obj.end_time - obj.start_time;
            end
        end
    end
    
    methods (Static)
         function obj = load_run(run_file)
            filename = [pwd '\data\runs\' run_file '.mat'];
            try
                loaded_obj = load(filename, 'run_case');
                obj = loaded_obj.run_case;
            catch
                error('Supplied file has no property: run_case')
            end
        end
    end
end

Appendices/MATLAB/main.m
clear all;
close all;
clc;

gamma=1.4; %Ratio of Specific Heats (Air)

%% Importing Tables from Nozzle1.csv and Nozzle2.csv

global NoZ1 NoZ2

NoZ1.data=readtable('Nozzle1.csv'); %NoZ1 = Nozzle1
NoZ2.data=readtable('Nozzle2.csv'); %NoZ2 = Nozzle2

%% Importing Experimental Data

NoZ1.ExperimentSUP=readtable('2017-02-17_12-21-48.txt'); 
NoZ1.ExperimentSUB=readtable('2017-02-17_12-24-22.txt');
NoZ2.Experiment1=readtable('2017-02-17_12-37-39.txt');
NoZ2.Experiment2=readtable('2017-02-17_12-31-42.txt');
NoZ2.Experiment3=readtable('2017-02-17_12-34-20.txt');

%% Theoretical Supersonic Flow for First Nozzle [Choked Flow Conditions At=A*]

NoZ1.SUP.A=NoZ1.data.A;
NoZ1.SUP.MACH=zeros(length(NoZ1.SUP.A),1);
NoZ1.SUP.P=zeros(length(NoZ1.SUP.A),1);
for i=1:length(NoZ1.SUP.A);
    if i<6
        [NoZ1.SUP.MACH(i), ~, NoZ1.SUP.P(i), ~, ~]=flowisentropic(gamma, NoZ1.SUP.A(i), 'sub');
    else
        [NoZ1.SUP.MACH(i), ~, NoZ1.SUP.P(i), ~, ~]=flowisentropic(gamma, NoZ1.SUP.A(i), 'sup');
    end
end

%% Theoretical Subsonic Flow for First Nozzle [Non-Choked Flow Conditions At=/=A*]

PRatio=NoZ1.ExperimentSUB.P_Pt(1); % Measured Pressure Ratio at First Pressure Measurement

[~, ~, ~, ~, NoZ1.SUB.A_star]=flowisentropic(gamma, PRatio, 'pres');
NoZ1.SUB.A_correction=NoZ1.SUB.A_star*(1/NoZ1.data.A(1)); %A(x0)/A_star * A_t/A(x0) = A_t/A_star
NoZ1.SUB.A=NoZ1.SUB.A_correction.*NoZ1.data.A;

for i=1:length(NoZ1.SUB.A);
    [NoZ1.SUB.MACH(i), ~, NoZ1.SUB.P(i), ~, ~]=flowisentropic(gamma, NoZ1.SUB.A(i), 'sub');
end

%% Obtaining Experimental Mach Number using Isentropic Relations

NoZ1.SUP.MACH_EXP=zeros(length(NoZ1.ExperimentSUP.mm),1);
NoZ1.SUB.MACH_EXP=zeros(length(NoZ1.ExperimentSUP.mm),1);

for i=1:length(NoZ1.ExperimentSUP.mm)
        [NoZ1.SUP.MACH_EXP(i), ~, ~, ~, ~]=flowisentropic(gamma, NoZ1.ExperimentSUP.P_Pt(i), 'pres');
        [NoZ1.SUB.MACH_EXP(i), ~, ~, ~, ~]=flowisentropic(gamma, NoZ1.ExperimentSUB.P_Pt(i), 'pres');
end

%% Calculating Area Ratio at Shock Location

NoZ2.x_shock=[390 530 630];
[hk2,dh,h0]=meter2geo(15,10); %Input Experiment Meter Values Here (meter4,meter5)
A_ratio=(dh*NoZ2.x_shock(2:3)+h0)/hk2;

for i=1:length(A_ratio)
    [~, ~, NoZ2.SUB.Pe3(i), ~, ~]=flowisentropic(gamma, A_ratio(i), 'sub');
    [NoZ2.SUP.MACH_XSHOCK(i), ~, NoZ2.SUP.Pe6(i), ~, ~]=flowisentropic(gamma, A_ratio(i), 'sup');
    [~, ~, NoZ2.SUP.Pe5(i), ~, ~, ~, ~]=flownormalshock(gamma, NoZ2.SUP.MACH_XSHOCK(i), 'mach');
    NoZ2.SUP.Pe5(i)=NoZ2.SUP.Pe5(i)*NoZ2.SUP.Pe6(i);
end

%% Plotting Fiures

%Part I:

Margin=0.125; %Control Figure Margins
LabelSize=9; %Control Label Text Size
AR=[8 5]; %Aspect Ratio
NF=0.01; %Distance to Nudge Plot to Compensate for Axis Labels

figure('Name','MachNoZ1');
hold on; grid on; grid minor;
line([65 65],[0 NoZ1.SUP.MACH(5)],'Color','k','LineStyle','- -'); %Drawing a Vertical Line to Indicate Throat Sonic Value
line([0 65],[1 1],'Color','k','LineStyle','- -'); %Drawing a Horizontal Line to Indicate Throat Sonic Value
line([0 194.8],[2.1 2.1],'Color','k','LineStyle','- -'); %Drawing a Horizontal Line to Indicate Throat Sonic Value
text(65.5,0.75,'$\leftarrow x_t = 65 \left[ mm \right]$','Interpreter','LaTex','FontSize',8)
text(50,1.05,'M = 1','Interpreter','LaTex','FontSize',8)
text(120,2.15,['$M_{e} =$ ' num2str(NoZ1.SUP.MACH(26))],'Interpreter','LaTex','FontSize',8)
line1=plot(NoZ1.data.x,NoZ1.SUB.MACH,'LineWidth',1);
line2=plot(NoZ1.data.x,NoZ1.SUP.MACH,'LineWidth',1);
line3=plot(NoZ1.ExperimentSUB.mm,NoZ1.SUB.MACH_EXP,'Color',[0 174 255]/255,'marker','o','MarkerFaceColor','w','MarkerSize',3,'LineWidth',1);
line4=plot(NoZ1.ExperimentSUP.mm,NoZ1.SUP.MACH_EXP,'Color',[221 135 51]/255,'marker','o','MarkerFaceColor','w','MarkerSize',3,'LineWidth',1);
axis([44.8 194.8 0 2.5])
xlabel('Position [mm]','fontsize',12,'Interpreter','LaTex');
ylabel('Mach Number [-]','fontsize',12,'Interpreter','LaTex');
legend([line1,line3,line2,line4],{'Subsonic [Theory]','Subsonic [EXP1A]','Supersonic [Theory]','Supersonic [EXP2A]'},'Location','East'); %Creating Legend
set(gca,... %Formatting Axis Text
    'XMinorTick','on',...
    'YMinorTick','on',...
    'FontSize',LabelSize/1.5,...
    'TickLabelInterpreter','LaTex',...
    'LabelFontSizeMultiplier',1.5,...
    'Position',[((Margin+NF)/2) ((0+((AR(1)/AR(2))*(Margin)))/2) (1-Margin) (1-((AR(1)/AR(2))*Margin))]);
set(gcf, 'InvertHardCopy', 'off');
set(gcf, 'PaperPosition', [0 0 AR(1) AR(2)]); %Position plot at left hand corner with width 6 and height 5.
set(gcf, 'PaperSize', [AR(1) AR(2)]); %Set the paper to have width 6 and height 5.

figure('Name','PressureNoZ1');
hold on; grid on; grid minor;
line([65 65],[0 NoZ1.SUB.P(5)],'Color','k','LineStyle','- -'); %Drawing a Vertical Line to Indicate Throat Sonic Value
line([0 65],[0.528 0.528],'Color','k','LineStyle','- -'); %Drawing a Horizontal Line to Indicate Throat Sonic Value
text(65.5,0.1,'$\leftarrow x_t = 65 \left[ mm \right]$','Interpreter','LaTex','FontSize',8)
text(66,0.528,'$\leftarrow \frac{p}{p_t}=0.528$','Interpreter','LaTex','FontSize',8)
line1=plot(NoZ1.data.x,NoZ1.SUB.P,'LineWidth',1);
line2=plot(NoZ1.data.x,NoZ1.SUP.P,'LineWidth',1);
line3=plot(NoZ1.ExperimentSUB.mm,NoZ1.ExperimentSUB.P_Pt,'Color',[0 174 255]/255,'marker','o','MarkerFaceColor','w','MarkerSize',3,'LineWidth',1);
line4=plot(NoZ1.ExperimentSUP.mm,NoZ1.ExperimentSUP.P_Pt,'Color',[221 135 51]/255,'marker','o','MarkerFaceColor','w','MarkerSize',3,'LineWidth',1);
xlabel('Position [mm]','Interpreter','LaTex','Color','k');
ylabel('Pressure Ratio $\left[ \frac{p}{p_{t}} \right]$','Interpreter','LaTex','Color','k');
legend([line1,line3,line2,line4],{'Subsonic [Theory]','Subsonic [EXP1]','Supersonic [Theory]','Supersonic [EXP2]'},'Location','East'); %Creating Legend
axis([44.8 194.8 0 1]) %Setting Axis Limits
set(gca,... %Formatting Axis Text
    'XMinorTick','on',...
    'YMinorTick','on',...
    'FontSize',LabelSize/1.5,...
    'TickLabelInterpreter','LaTex',...
    'LabelFontSizeMultiplier',1.5,...
    'Position',[((Margin+NF)/2) ((0+((AR(1)/AR(2))*(Margin)))/2) (1-Margin) (1-((AR(1)/AR(2))*Margin))]);
set(gcf, 'InvertHardCopy', 'off');
set(gcf, 'PaperPosition', [0 0 AR(1) AR(2)]); %Position plot at left hand corner with width 6 and height 5.
set(gcf, 'PaperSize', [AR(1) AR(2)]); %Set the paper to have width 6 and height 5.

%Part II:

figure('Name','PressureNoZ2');
hold on; grid on; grid minor;
line1=plot(NoZ2.Experiment1.mm,NoZ2.Experiment1.P_Pt,'marker','o','MarkerFaceColor','w','MarkerSize',3,'LineWidth',1);
line2=plot(NoZ2.Experiment2.mm,NoZ2.Experiment2.P_Pt,'marker','o','MarkerFaceColor','w','MarkerSize',3,'LineWidth',1);
line3=plot(NoZ2.Experiment3.mm,NoZ2.Experiment3.P_Pt,'marker','o','MarkerFaceColor','w','MarkerSize',3,'LineWidth',1);
line4=line([NoZ2.x_shock(1) NoZ2.x_shock(1)],[0 1],'Color','k','LineStyle','-.');
line5=line([NoZ2.x_shock(2) NoZ2.x_shock(2)],[0 1],'Color','k','LineStyle','--');
line6=line([NoZ2.x_shock(3) NoZ2.x_shock(3)],[0 1],'Color','k');
point1=scatter(NoZ2.x_shock(2:3),NoZ2.SUB.Pe3,'*','MarkerEdgeColor',[0 0.60 0.50]);
point2=scatter(NoZ2.x_shock(2:3),NoZ2.SUP.Pe5,'*','MarkerEdgeColor',[0.80 0.40 0]);
point3=scatter(NoZ2.x_shock(2:3),NoZ2.SUP.Pe6,'*');
xlabel('Position [mm]','Interpreter','LaTex','Color','k');
ylabel('Pressure Ratio $\left[ \frac{p}{p_{t}} \right]$','Interpreter','LaTex','Color','k');
legend([line1,line2,line3,line4,line5,line6,point1,point2,point3],{'Experiment 3A','Experiment 4A','Experiment 5A','Shock at Throat','Shock at $x=530$ [mm]','Shock at $x=630$ [mm]','Maximum Pressure Ratio [$p_{e3}/{p_t}$]','Pressure Ratio After Norm. Shock [$p_{e5}/{p_t}$]','Minumum Pressure Ratio [$p_{e6}/{p_t}$]'},'Location','West','Interpreter','LaTex'); %Creating Legend
set(gca,... %Formatting Axis Text
    'XMinorTick','on',...
    'YMinorTick','on',...
    'FontSize',LabelSize/1.5,...
    'TickLabelInterpreter','LaTex',...
    'LabelFontSizeMultiplier',1.5,...
    'Position',[((Margin+NF)/2) ((0+((AR(1)/AR(2))*(Margin)))/2) (1-Margin) (1-((AR(1)/AR(2))*Margin))]);
set(gcf, 'InvertHardCopy', 'off');
set(gcf, 'PaperPosition', [0 0 AR(1) AR(2)]); %Position plot at left hand corner with width 6 and height 5.
set(gcf, 'PaperSize', [AR(1) AR(2)]); %Set the paper to have width 6 and height 5.

clear vars Margin LabelSize AR NF

%% Saving/Overwriting Figures in the Images Folder as a .pdf

choice=questdlg('Would you like to close and save all figures to ../Figures?',...
    'Figure Save Dialog', ...
    'Yes','Just Save','Specify Directory','Specify Directory');
switch choice
    case 'Yes'
        if exist('Figures','dir')==0
        mkdir('Figures')
        end
        cd('Figures')
        b1=waitbar(0,'1','Name','Please Wait');
        H=gcf;
        i_prime=H.Number;
        for i=1:i_prime
            waitbar(i/i_prime,b1,sprintf('Saving Figure (%d/%d)',i,i_prime))
            saveas(i, get(i,'Name'), 'pdf')
        end
        cd('../')
        close all
        delete(b1)
        b2=msgbox('Operation Completed','Success');
    case 'Just Save'
        if exist('Figures','dir')==0
        mkdir('Figures')
        end
        cd('Figures')
        b1=waitbar(0,'1','Name','Please Wait');
        H=gcf;
        i_prime=H.Number;
        for i=1:i_prime
            waitbar(i/i_prime,b1,sprintf('Saving Figure (%d/%d)',i,i_prime))
            saveas(i, get(i,'Name'), 'pdf')
        end
        cd('../')
        delete(b1)
        b2=msgbox('Operation Completed','Success');
    case 'Specify Directory'
        old_dir=cd;
        new_dir=uigetdir('','Select Figure Saving Directory');
        cd(new_dir)
        b1=waitbar(0,'1','Name','Please Wait');
        H=gcf;
        i_prime=H.Number;
        for i=1:i_prime
            waitbar(i/i_prime,b1,sprintf('Saving Figure (%d/%d)',i,i_prime))
            saveas(i, get(i,'Name'), 'pdf')
        end
        cd(old_dir)
        close all
        delete(b1)
        b2=msgbox('Operation Completed','Success');
end